client/src/rpc_response.rs

```diff
@@ -101,6 +101,16 @@ pub struct SlotInfo {
     pub root: Slot,
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase", tag = "type")]
+pub enum SlotUpdate {
+    OptimisticConfirmation { slot: Slot, timestamp: u64 },
```

Comment:
How about errored blocks? Missing I guess we know if there is just no update for a given slot.

Reply:
hmmm yeah if we sent all the roots, then you could tell which ones are missing due to wrong fork/errors. If we want to distinguish between wrong fork/replayed error we could add an extra errored block notification